R/zchunk_L110.water.demand.primary.R

```diff
@@ -33,45 +33,78 @@ module_water_L110.water.demand.primary_DISABLED <- function(command, ...) {
     resource_water_share <- get_data(all_data, "water/resource_water_share")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are `merge` and/or 'match' calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Merge-and-Match
-    # ===================================================
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # Get water consumption (m^3/TJ) by fuel and supply sector
+    A227.resource_water_coef_mapping %>%
+      left_join(resource_water_data, by = c("fuel", "subsector", "technology")) %>%
+      select(fuel, supplysector, water.coefficient.m3.per.TJ) %>%
+      arrange(fuel) %>%
+      select(-fuel) %>%
+      unique() -> L110.global_water_cons_coef
+
+    # Create tibble for all regions and supply sectors
+    GCAM_region_names %>%
+      merge(L110.global_water_cons_coef) %>%
```

Comment:
Thought you might raise this! None of the dplyr join functions work in this instance because there's no common variable. All merge is doing here is repeating L110.global_water_cons_coef across each of the region names. Is there a neater way to do this using dplyr??